internal/providers/provider.go

```diff
@@ -38,15 +38,19 @@ type Interface interface {
 	// configuration values.
 	ValidateDataResourceConfig(ValidateDataResourceConfigRequest) ValidateDataResourceConfigResponse
 
-	// UpgradeResourceState is called when the state loader encounters an
-	// instance state whose schema version is less than the one reported by the
-	// currently-used version of the corresponding provider, and the upgraded
-	// result is used for any further processing.
-	UpgradeResourceState(UpgradeResourceStateRequest) UpgradeResourceStateResponse
+	// MoveResourceState requests that the given resource data be moved from one
+	// type to another, potentially between providers as well.
+	MoveResourceState(MoveResourceStateRequest) MoveResourceStateResponse
+
+	// CallFunction requests that the given function is called and response returned.
+	CallFunction(CallFunctionRequest) CallFunctionResponse
```

Comment:
I don't remember the details, but we do allow custom functions to be called. I assume they are part of the configuration process so they will be unavailable here?

Reply:
I am also confused that we can `CallFunction` but not `GetFunctions` with `Unconfigured`. Comment would be helpful here.

Reply:
CallFunction is a weird one, it can call both unconfigured and configured functions in OpenTofu-land.  I can update the documentation to clarify this quirk.